src/js/base/ProgressCircle.tsx

```diff
@@ -0,0 +1,223 @@
+import { Indicator, Root } from "@radix-ui/react-progress";
+import React from "react";
+import styled, { keyframes } from "styled-components";
+import { getColor, sizes } from "../app/theme";
+import { JobState } from "../jobs/types";
+
+/**
+ * Calculate the stroke width based on the total size of the progress circle
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @returns  The stroke width in pixels
+ */
+
+function getStrokeWidth(size: number): number {
+    return size / 5;
+}
+
+/**
+ * Calculate the radius of the circle based on the total size and stroke width
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @returns The stroke width in pixels
+ */
+function getRadius(size: number): number {
```

Comment:
This might be a good idea as it could get confused with theme helpers for getting the border radius.
```suggestion
function getProgressRadius(size: number): number {
```